Forms: Fix jwt form submission validation

enejb opened the pull request:
Fixes FORMS-NEW

Fixes form submissions with empty fields on the backend. Currently we limit form submission for fields via JS but not for form submissions. 

## Proposed changes:
* Validate form fields before processing the form. 
* Check that at least one field is valid and has a value. 

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?


## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
Disable JS in your browser and submit the empty form. 
You should see an error. 
If the fields are required you should see the field specific errors. 

You should be able to submit the form as expected if the form is filled out correctly without JS. 

In future PRs it would be good to display the error message in the fields as expected.

enejb committed:
Add form validation and error handling improvements

Introduces a validate() method to Contact_Form for field validation and empty form checks, refactors error handling to use static errors, and updates error message formatting. Updates related tests to cover new validation logic and error scenarios.

enejb committed:
changelog

enejb committed:
Fix Phan error

enejb committed:
Add more error messages tests

enejb committed:
Test empty form submission

CGastrell committed:
trim values when checking if field has value

CGastrell committed:
if a field is marked as error, skip validation

CGastrell committed:
remove field validation, this is ran from the form

enejb committed:
Fix the late processing

enejb committed:
Fix Phan Error!

enejb committed:
Improve checkbox-multiple field validation in contact form

Enhanced validation logic for 'checkbox-multiple' fields to ensure at least one valid selection is made and that selected options are among allowed values. Updated related tests to cover new validation scenarios and adjusted test utility to handle array values in POST data.

enejb committed:
Revert "remove field validation, this is ran from the form"

This reverts commit af8f6fb2a4832fd48cb597c5f1e9343da8f211e9.

enejb committed:
Add JWT check to contact form submission

Added a check to ensure 'jetpack_contact_form_jwt' is set in POST data before processing contact form submissions. This improves security by requiring a JWT for form actions.
This prevents the double varification

CGastrell reviewed (dismissed):
This tested well, we do have to followup on first submit not showing errors, but it already improves how we handle empty submits.

enejb committed:
Fix validation for non-required fields with values

Updates the contact form field validation logic to ensure that non-required fields are validated if they have a value. Adds corresponding test coverage for a non-required email field to verify correct validation behavior.

CGastrell committed:
add href to go back link for proper no JS fallback

CGastrell committed:
return false from goBack link handler so it doesn't trigger default link behavior

CGastrell committed:
proper event prevention

CGastrell committed:
let back link fallback take you to the form position
